blueprint/automation.md

```diff
@@ -0,0 +1,170 @@
+2. Certificate creation
+3. Configuration deployment
+
+If the certificate creation part is skipped, the configuration file (MOF file) will contain credentials in **clear text**. Also the `CertificateFile` value within `configurationdata.psd1` would require removal.
+
+#### Configuration file development
+
+The configuration files developed in accordance with the blueprint are available below. The configuration data file is required no matter the configuration script used.
+
+##### Configuration data file
+
+The required `configurationdata.psd1` file can be downloaded below:
+
+* [configurationdata.psd1](/assets/files/automation/configurationdata.txt)
+
+##### Blueprint configuration scripts
+
+Blueprint configuration scripts are available below:
+
+* [identity_dsc.ps1](/assets/files/automation/identity_dsc.txt)
+
+#### Certificate creation
+
+The process to create a certificate is as follows:
+
+1. Open a PowerShell console as administrator
+2. Install and import the PowerShell module `microsoft365dsc` using the following command:
+```Powershell 
+install-module microsoft365dsc -allowclobber -force
+import-module microsoft365dsc
+```
+3. In PowerShell navigate to the directory containing the Blueprint configuration PS1 file and the `configurationdata.psd1` file
+4. Create a Desired State Configuration local configuration manager certificate:
+```Powershell 
+Set-M365DSCAgentCertificateConfiguration
+```
+5. Validate that a certificate was created and assigned to the local configuration manager by matching the output of step 4 with `CertificateID` from the following:
+```Powershell 
+Get-DscLocalConfigurationManager
+```
+6. Export the certificate into the folder containing the Blueprint configuration PS1 file and the `configurationdata.psd1` file
+```Powershell
+$certificatethumbprint = Get-DscLocalConfigurationManager | select CertificateID
+$certpath =  "cert:\localmachine\My\" + $certificatethumbprint.CertificateID
+$cert = Get-ChildItem -path $certpath
+Export-Certificate -Cert $cert -FilePath .\M365dsc_DER.cer
+certutil -encode M365dsc_DER.cer m365dsc.cer
+remove-item .\M365dsc_DER.cer
+```
+
+#### Configuration deployment
+
+The process to deploy the configuration is as follows:
+
+1. Open a PowerShell console as administrator
+2. Install and import the PowerShell module microsoft365dsc using the following command:
+```Powershell 
+install-module microsoft365dsc -allowclobber -force
+import-module microsoft365dsc
+```
+3. In PowerShell navigate to the directory containing the Blueprint configuration PS1 file and the `configurationdata.psd1` file
+4. In the PowerShell console, initiate the Blueprint configuration PS1 and supply the required values. The required values are located in the parameters section of the `Blueprint Configuration Script file`. The following is an example for the identity configuration. The required values are: Global Admin credentials; agency name; agency prefix; and trusted IPs (in CIDR format).
```

Comment:
Just for clarity, `Blueprint Configuration Script file` means `identity_dsc.ps1`?

Reply:
Correct. I was aiming to make the procedure as generic as possible to prevent future duplication. I have now added an i.e. identity_dsc.ps1 - this should address any confusion.

Let me know if it needs changing